packages/vow/test/retrier.test.js

```diff
@@ -0,0 +1,130 @@
+import '@agoric/swingset-liveslots/tools/prepare-test-env.js';
+import test from '@endo/ses-ava/prepare-endo.js';
+
+import { Fail } from '@endo/errors';
+import { passStyleOf } from '@endo/pass-style';
+import { eventLoopIteration } from '@agoric/internal/src/testing-utils.js';
+import { makeHeapZone } from '@agoric/zone/heap.js';
+import { makeVirtualZone } from '@agoric/zone/virtual.js';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+import { reincarnate } from '@agoric/swingset-liveslots/tools/setup-vat-data.js';
+import { makeUpgradeDisconnection } from '@agoric/internal/src/upgrade-api.js';
+
+import { prepareVowTools } from '../vat.js';
+import { prepareRetrierTools } from '../src/retrier.js';
+import { isVow } from '../src/vow-utils.js';
+
+/**
+ * @import {Zone} from '@agoric/base-zone'
+ */
+
+/** @type {ReturnType<typeof reincarnate>} */
+let incarnation;
+
+const annihilate = () => {
+  incarnation = reincarnate({ relaxDurabilityRules: false });
+};
+
+const getBaggage = () => {
+  return incarnation.fakeVomKit.cm.provideBaggage();
+};
+
+const nextLife = () => {
+  incarnation = reincarnate(incarnation);
+};
+
+/**
+ * @param {any} t
+ * @param {Zone} zone
+ */
+const retrierPlay1 = async (t, zone) => {
+  const vowTools = prepareVowTools(zone);
+  const { retry, retrierAdmin } = prepareRetrierTools(zone, vowTools);
+
+  const makeBob = zone.exoClass('Bob', undefined, count => ({ count }), {
+    foo(carol) {
+      const { state } = this;
+      state.count += 1;
+      carol.ping(state.count);
+      if (state.count < 102) {
+        throw makeUpgradeDisconnection('emulated upgrade1', state.count);
+      } else {
+        t.log('postponed at', state.count);
+        return new Promise(() => {}); // never resolves
+      }
+    },
+  });
+  const bob = makeBob(100);
+  const carol = zone.exo('carol', undefined, {
+    ping(count) {
+      t.log('ping at', count);
+    },
+  });
+  const v = zone.makeOnce('v', () => retry(bob, 'foo', [carol]));
```

Comment:
Note that the call to `retry` only happens in the first incarnation, but it continues retrying into the second without further preparation.